[BUGFIX] Fix Two Issues With Song Text on Freeplay Capsules

VioletSnowLeopard opened the pull request:
<!-- Please read the Contributing Guide (https://github.com/FunkinCrew/Funkin/blob/main/docs/CONTRIBUTING.md) before submitting this PR. -->

<!-- Does this PR close any issues? If so, link them below. -->
## Linked Issues
Fixes #4876. Fixes #2659.
<!-- Briefly describe the issue(s) fixed. -->
## Description
When switching song lists (changing filters or difficulties) from a song not in the new list, its text would remain highlighted when you switched back to the list it is in. This PR fixes the issue by setting the `selected` value of the capsule to `false` before the song list might change.

I also made an adjustment in `SongMenuItem.hx` to fix the squashed text. A detailed description of the change is in a comment below.
<!-- Include any relevant screenshots or videos. -->
## Screenshots/Videos
Video of song text not being highlighted:

https://github.com/user-attachments/assets/cd647ce0-f5e1-4034-9aae-8198eb12759d

After the 0.6.4 update, there were three scenarios I found that would always result in a capsule having squished text which is shown below:

https://github.com/user-attachments/assets/9141f79a-1276-48a3-bf75-4cede81fb4e3

Here, you can see these same scenarios no longer produce squished text:

https://github.com/user-attachments/assets/ac74250e-aaf3-4197-8679-b56fe8bcd861

VioletSnowLeopard commented:
> Could it really be? The one to fix the squashed text?

Would be funny since it wasn't the issue I was originally trying to fix.

Lasercar commented:
Now we just have to figure out why the capsules started sometimes getting that weird y offset in 0.6

VioletSnowLeopard commented:
> > Could it really be? The one to fix the squashed text?
> 
> Hundrec. I Need A Compiled Build. I Need To Test This. Yeah I Can't Be Bothered To Waste 20GB On My Storage Space For Funkin'

If you want a compiled build for Windows, I can zip the one I have. You can download it [here](https://drive.google.com/file/d/1t9OhR12ZI5OAPeazWjZ4dDaWnAf9hxYs/view?usp=drive_link) if you want to try it.

JVNpixels reviewed (approved):
thats it? just 2 lines? thats all it takes to fix 2 of my annoyances of the freeplay menu?

Hundrec commented:
aw

![image](https://github.com/user-attachments/assets/ae4e9552-90ef-4c32-9fb9-071fa4699098)

I got this by entering and exiting Stress on Hard

Lasercar commented:
> aw
> 
> I got this by entering and exiting Stress on Hard
> 
> (This does fix the highlighting though)

We've been foiled again by the squashed text

VioletSnowLeopard commented:
> aw
> I got this by entering and exiting Stress on Hard
> 
> (This does fix the highlighting though)

Are you able to get this consistently? Did you do anything before entering Stress? Because I can't seem to get this to occur.

trayfellow commented:
> > aw
> > I got this by entering and exiting Stress on Hard
> > (This does fix the highlighting though)
> 
> We've been foiled again by the squashed text

My worst enemy.... just like The Lich from Adventure Time, this stupid thing just wont go away no matter what

VioletSnowLeopard commented:
With this PR, I have found a way to reliably reproduce the squashed text on one of the capsules. I can try to figure out how it happens through debugging later.

Hundrec commented:
I left a comment under the squashed text issue outlining how to reproduce the bug by switching variations. Hopefully that's helpful!

VioletSnowLeopard commented:
Using debugging, I have found the problematic function. Whenever a capsule has squashed text, its x scale has a value of 0.4 and its y scale has a value of 1.4. The function that sets the text's scale values to this is `textAppear`. It seems that sometimes the second `FlxTimer` fails to work so the text's scales aren't reset to 1.

The purpose of `textAppear` is to make the text do a little jump back and forth animation. You can see this whenever you hit confirm on a song. Apparently this also happens to all capsules whenever changing difficulties or entering the menu, although I've never noticed it. I believe it happens at a point before the capsules are visible again.

Considering `textAppear` is the function that causes this issue and its functionality isn't ever noticeable aside from hitting confirm on a song, I removed the call to the function in `setVisibleGrp`. To allow it to still work when hitting confirm on a song, I added a call to it inside the `confirm` function in `SongMenuItem`.

AbnormalPoof commented:
It was that simple??

Hundrec reviewed (approved):
I think you've done it!
You've finally freed us from this curse...

VioletSnowLeopard commented:
> I think you've done it! You've finally freed us from this curse...

The cursed function `textAppear`

AbnormalPoof reviewed (approved):
peak

JackXson-Real reviewed (approved):
Merge this PR immediately! 🗣️ 🔥

Kade-github reviewed (approved):
very good, will be merged in the next update.
